### PR TITLE
Added the `profile` workflow including methods, classes, widgets and screens

### PR DESCRIPTION
**Pull request by ManasMalla**
1. `VolunteerModel` and `Volunteer` entity that work to create, store and fetch data regarding volunteers from/to database
2. `SpeakerModel` and `Speaker` entity work to create, store and fetch speakers from/to database
3. Added utils: `getAvatar` to get the social profile overlay and `capitalize` to capitalize sentences for `Sentence Case`
4. Added enum `UserRole` that encompasses `attendee`, `volunteer`, `organizer` and `speaker`
5. Added constants for color scheme
6. Added `widgets` that together build up the `profile_screen.dart` and their constituent widgets
7. Added the success bottom sheet when creating a profile
8. Added the respective navigations for the `create-profile` workflow

### CODE REVIEW

**Comment by ManasMalla** [comment]
Works to solve tasks 1, 2 and 3 and 5 of #11

**Comment by sampathbalivada** [comment]
TLDR;

Approving the PR

File: lib/data/models/speaker_model.dart [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:nowingoogle/data/models/user_model.dart';
+import 'package:nowingoogle/domain/entities/speaker.dart';
+import 'package:nowingoogle/domain/enums/user_roles.dart';
+import 'package:nowingoogle/presentation/pages/constants/color_scheme.dart';
+
+class SpeakerModel extends UserModel {
```
Comment (Harryalways317) [inline lib/data/models/speaker_model.dart:7]: Will there be any user identifier in this model(uid) so that just with that we can show the user page in future, or can we reuse the email for the same case?

**Review by sampathbalivada — APPROVED** [review]
TLDR;

Approving the PR

File: lib/data/utils/capitalize.dart [hunk 2]
```diff
@@ -0,0 +1,8 @@
+String capitalize(String str) {
```
Comment (Harryalways317) [inline lib/data/utils/capitalize.dart:1]: we can add it as string extensions. we can create extensions.dart and reuse that for all extensions

https://dart.dev/language/extension-methods

File: lib/domain/entities/volunteer.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import 'package:nowingoogle/domain/entities/user.dart';
+import 'package:nowingoogle/domain/enums/user_roles.dart';
+import 'package:nowingoogle/presentation/pages/constants/color_scheme.dart';
+
+class Volunteer extends User {
+  final bool isPOC;
```
Comment (Harryalways317) [inline lib/domain/entities/volunteer.dart:6]: What does isPOC means?

**Review by Harryalways317 — APPROVED** [review]
LGTM!

File: lib/data/utils/capitalize.dart (hunk 2 above)
Comment (ManasMalla) [reply lib/data/utils/capitalize.dart:1]: Done.

File: lib/domain/entities/volunteer.dart (hunk 3 above)
Comment (ManasMalla) [reply lib/domain/entities/volunteer.dart:6]: isPOC is used to identify if the volunteer is a lead for his domain and will the responsibility of coordinating with his team and planning tasks, before taking it to the next hierarchy, i.e, organizers.

File: lib/data/models/user_model.dart [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:equatable/equatable.dart';
 import 'package:nowingoogle/domain/entities/user.dart';
+import 'package:nowingoogle/domain/enums/user_roles.dart';
 
 import '../../domain/enums/social_platforms.dart';
```
Comment (Vasanth-Korada) [inline lib/data/models/user_model.dart:6]: Always use absolute imports

File: lib/main.dart [hunk 5]
```diff
@@ -32,7 +34,9 @@ class MyApp extends StatelessWidget {
       ),
       initialRoute: 'splash',
       routes: {
-        "splash": (context) => const SplashPage(),
+        'splash': (context) => const SplashPage(),
```
Comment (Vasanth-Korada) [inline lib/main.dart:37]: Create a separate class AppRoutes, have a variable with Map<String, dynamic>. Move all these route over there

File: lib/presentation/pages/layouts/io_footer.dart [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+          child: SvgPicture.asset(
+            "assets/io-logo.svg",
+            height: 32,
+            colorFilter: ColorFilter.mode(primaryColor, BlendMode.srcIn),
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.only(right: 16, top: 16),
+          child: Text(
+            "© 2023. Manas Malla. All Rights Reserved",
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/io_footer.dart:23]: Keep all hard coded string in a separate strings.dart constant file

File: lib/presentation/pages/layouts/profile/profile_avatar_image.dart [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+class ProfileAvatarImage extends StatelessWidget {
+  const ProfileAvatarImage({
+    super.key,
+    required this.url,
+  });
+  final String url;
+  @override
+  Widget build(BuildContext context) {
+    return ClipOval(
+      child: Image.network(
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_avatar_image.dart:12]: If possible write a custom NetworkImage widget which will handle failure case, image caching, loading placeholder e.t.c Not a priority as of now

File: lib/presentation/pages/layouts/profile/profile_banner.dart [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  const ProfileBanner({
+    super.key,
+    required this.primaryColor,
+  });
+
+  final Color primaryColor;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_banner.dart:13]: Use SizedBox and ColoredBox instead of Container since container is a heavy widget

File: lib/presentation/pages/layouts/profile/profile_header.dart [hunk 9]
```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import './profile_banner.dart';
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_header.dart:2]: Use absolute imports

File: lib/presentation/pages/layouts/profile/profile_image.dart [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  final String image;
+  final String avatar;
+  const ProfileImage({super.key, required this.image, required this.avatar});
+
+  @override
+  Widget build(BuildContext context) {
+    return ClipOval(
+      child: SizedBox(
+        height: 128,
+        width: 128,
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_image.dart:13]: Try using AspectRatio widget wherever possible instead of explicitly mentioning width and height. 
aspectRatio = width / height

File: lib/presentation/pages/layouts/profile/profile_section.dart [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+import 'package:nowingoogle/domain/enums/user_roles.dart';
+import './profile_bio.dart';
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_section.dart:3]: Use absolute imports

File: lib/presentation/pages/layouts/profile/profile_title.dart [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+        ),
+        if (!isAttendee)
+          const SizedBox(
+            width: 8,
+          ),
+        if (!isAttendee)
+          const Icon(
+            Icons.verified,
+            size: 18,
+            color: Color(0xFF4285f4),
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_title.dart:30]: Create a app_colors.dart constant file and move all hard-coded colors to that file with a color name

You can use 
https://colors.artyclick.com/color-name-finder/
to find a color name for a given hex code

File: lib/presentation/pages/layouts/profile/profile_user_role.dart [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+  final bool isPOC;
+  final Color primaryColor;
+  final Color onPrimaryColor;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Container(
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/layouts/profile/profile_user_role.dart:24]: Use DecoratedBox widget instead Container wherever possible

File: lib/presentation/pages/profile_screen.dart [hunk 14]
```diff
@@ -0,0 +1,349 @@
...
+
+  @override
+  State<ProfileScreen> createState() => _ProfileScreenState();
+}
+
+class _ProfileScreenState extends State<ProfileScreen> {
+  var isEditing = false;
+  @override
+  Widget build(BuildContext context) {
+    final User user = ModalRoute.of(context)?.settings.arguments as User;
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/profile_screen.dart:27]: Please move all of these outside of build method and use appropriate lifecycle methods

File: lib/presentation/pages/profile_screen.dart [hunk 15]
```diff
@@ -0,0 +1,349 @@
...
+              },
+              label: const Text("Edit Profile"),
+              icon: const Icon(Icons.edit_outlined),
+            )
+          : null,
+      body: AnimatedSwitcher(
+        duration: const Duration(milliseconds: 300),
+        child: SingleChildScrollView(
+          child: LayoutBuilder(builder: (context, constraints) {
+            return SizedBox(
```
Comment (Vasanth-Korada) [inline lib/presentation/pages/profile_screen.dart:55]: Please use Form widget whenever you are building a form.
